### PR TITLE
Drop osa and jabberd from server

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

osa-dispatcher and jabberd are not supported anymore with master.
Drop setting them up. Remove debugging code and replace a config value used to determine the own hostname.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fix partly https://github.com/SUSE/spacewalk/issues/18882

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by cbosdo — COMMENTED** [review]
LGTM, just left a few cosmetic comments

File: python/spacewalk/server/handlers/xmlrpc/registration.py [hunk 1]
```diff
@@ -1233,51 +1233,14 @@ def finish_message(self, system_id):
...
-    _query_get_dispatchers = rhnSQL.Statement("""
-        select jabber_id from rhnPushDispatcher
-    """)
-
-    def _get_dispatchers(self):
-        h = rhnSQL.prepare(self._query_get_dispatchers)
-        h.execute()
-        return [x['jabber_id'] for x in h.fetchall_dict() or []]
-
     def register_osad(self, system_id, args={}):
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply python/spacewalk/server/handlers/xmlrpc/registration.py:1236]: I would for now. At some later point we might decide to drop the whole file.
For now and osad would not crash, just not work anymore.

File: java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public class GeneralConfigAction extends BaseConfigAction {
 
 
     private static final String[] STRING_CONFIG_ARRAY = {"traceback_mail",
-        "server.jabber_server", "server.satellite.http_proxy",
+        "java.hostname", "server.satellite.http_proxy",
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java]: fixed

File: java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java [hunk 3]
```diff
@@ -157,9 +157,9 @@ public ActionForward execute(ActionMapping mapping,
                                 translateFormPropertyName("server.satellite.http_proxy_password_confirm"),
                                 configValue);
                     }
-                    else if (configKey.equals("server.jabber_server")) {
+                    else if (configKey.equals("java.hostname")) {
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java]: fixed

File: java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java [hunk 4]
```diff
@@ -157,9 +157,9 @@ public ActionForward execute(ActionMapping mapping,
...
                         form.set(
-                                translateFormPropertyName("server.jabber_server"),
+                                translateFormPropertyName("java.hostname"),
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java]: fixed

File: java/buildconf/test/rhn.conf.postgresql-example [hunk 5]
```diff
@@ -79,6 +79,8 @@ server.satellite.http_proxy =
 server.satellite.http_proxy_username =
 server.satellite.http_proxy_password =
 
+java.hostname = linux.box.top
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/buildconf/test/rhn.conf.postgresql-example]: changed

File: java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java [hunk 6]
```diff
@@ -123,7 +123,7 @@ public ActionForward execute(ActionMapping mapping,
                 else {
                     String value = (String)
                         form.get(translateFormPropertyName(configKey));
-                    if (configKey.equals("server.jabber_server")) {
+                    if (configKey.equals("java.hostname")) {
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/frontend/action/satellite/GeneralConfigAction.java]: fixed

**Pull request merged**